From 23e444e7519d661e17d11c8effea038c1ef3875f Mon Sep 17 00:00:00 2001
From: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>
Date: Thu, 4 Feb 2016 13:04:14 +0200
Subject: [PATCH] iproute2: fix build for older systems

Upstream-Status: Inapropriate

Signed-off-by: Ioan-Adrian Ratiu <adrian.ratiu@ni.com>
---
 ip/ipnetns.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/ip/ipnetns.c b/ip/ipnetns.c
index be0c473..dcda5d8 100644
--- a/ip/ipnetns.c
+++ b/ip/ipnetns.c
@@ -23,6 +23,19 @@
 #include "ip_common.h"
 #include "namespace.h"
 
+/* sys/mount.h may be out too old to have these */
+#ifndef MS_REC
+#define MS_REC		16384
+#endif
+
+#ifndef MS_SLAVE
+#define MS_SLAVE	(1 << 19)
+#endif
+
+#ifndef MS_SHARED
+#define MS_SHARED	(1 << 20)
+#endif
+
 static int usage(void)
 {
 	fprintf(stderr, "Usage: ip netns list\n");
-- 
2.7.0

