From aba8b39375cfce6ab67292707ad82891435aa2b9 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Sun, 1 Sep 2013 14:16:04 -0500
Subject: [PATCH] crypto/ts: Implement def_time_cb() on VxWorks

Upstream-Status: Inappropriate [VxWorks support]

---
 crypto/ts/ts_rsp_sign.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/crypto/ts/ts_rsp_sign.c b/crypto/ts/ts_rsp_sign.c
index d55e903..39262ef 100644
--- a/crypto/ts/ts_rsp_sign.c
+++ b/crypto/ts/ts_rsp_sign.c
@@ -61,7 +61,11 @@
 #include "o_time.h"
 
 #if defined(OPENSSL_SYS_UNIX)
+#ifndef VXWORKS_PPC
 # include <sys/time.h>
+#else
+# include <time.h>
+#endif
 #endif
 
 #include <openssl/objects.h>
@@ -116,6 +120,10 @@ static ASN1_INTEGER *def_serial_cb(struct TS_resp_ctx *ctx, void *data)
 static int def_time_cb(struct TS_resp_ctx *ctx, void *data,
                        long *sec, long *usec)
 {
+#ifdef VXWORKS_PPC
+    struct timespec timeSpec;
+    clock_gettime(CLOCK_REALTIME, &timeSpec);
+#else
     struct timeval tv;
     if (gettimeofday(&tv, NULL) != 0) {
         TSerr(TS_F_DEF_TIME_CB, TS_R_TIME_SYSCALL_ERROR);
@@ -124,9 +132,15 @@ static int def_time_cb(struct TS_resp_ctx *ctx, void *data,
         TS_RESP_CTX_add_failure_info(ctx, TS_INFO_TIME_NOT_AVAILABLE);
         return 0;
     }
+#endif
     /* Return time to caller. */
+#ifdef VXWORKS_PPC
+    *sec = timeSpec.tv_sec;
+    *usec = timeSpec.tv_nsec / 1000;
+#else
     *sec = tv.tv_sec;
     *usec = tv.tv_usec;
+#endif
 
     return 1;
 }
