From 4d0611933b20dda2a05706adaf399f91e2b11ce1 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Thu, 2 Apr 2015 13:06:51 -0500
Subject: [PATCH] crypto: Implement OpenSSLDie() as abort() on Phar Lap ETS

Upstream-Status: Inappropriate [Phar Lap ETS support]

---
 crypto/cryptlib.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/cryptlib.c b/crypto/cryptlib.c
index 5fab45b..8c9e434 100644
--- a/crypto/cryptlib.c
+++ b/crypto/cryptlib.c
@@ -1005,7 +1005,7 @@ void OpenSSLDie(const char *file, int line, const char *assertion)
     OPENSSL_showfatal
         ("%s(%d): OpenSSL internal error, assertion failed: %s\n", file, line,
          assertion);
-#if !defined(_WIN32) || defined(__CYGWIN__)
+#if !defined(_WIN32) || defined(__CYGWIN__) || defined(PHARLAP)
     abort();
 #else
     /*
