From c11cf85dd1c76f66938a3fb5c90b1a76cc4af856 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Thu, 2 Apr 2015 14:38:30 -0500
Subject: [PATCH 09/13] crypto/des: Disable WIN_CONSOLE_BUG hack on Phar Lap
 ETS

Upstream-Status: Inappropriate [Phar Lap ETS support]
---
 crypto/des/read_pwd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/des/read_pwd.c b/crypto/des/read_pwd.c
index 514a7063b4..6c77585867 100644
--- a/crypto/des/read_pwd.c
+++ b/crypto/des/read_pwd.c
@@ -514,7 +514,7 @@ static int noecho_fgets(char *buf, int size, FILE *tty)
             break;
         }
     }
-#  ifdef WIN_CONSOLE_BUG
+#  if defined(WIN_CONSOLE_BUG) && !defined(PHARLAP)
     /*
      * Win95 has several evil console bugs: one of these is that the last
      * character read using getch() is passed to the next read: this is
-- 
2.13.1

