From 2c47b553794b8d760b2f1b24c3b2cc8ac1b5fd9a Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Thu, 2 Apr 2015 14:41:24 -0500
Subject: [PATCH 11/14] crypto/ui: Disable WIN_CONSOLE_BUG hack on Phar Lap ETS

Upstream-Status: Inappropriate [Phar Lap ETS support]
---
 crypto/ui/ui_openssl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/crypto/ui/ui_openssl.c b/crypto/ui/ui_openssl.c
index 5d66276..135a6b2 100644
--- a/crypto/ui/ui_openssl.c
+++ b/crypto/ui/ui_openssl.c
@@ -699,7 +699,7 @@ static int noecho_fgets(char *buf, int size, FILE *tty)
             break;
         }
     }
-# ifdef WIN_CONSOLE_BUG
+# if defined(WIN_CONSOLE_BUG) && !defined(PHARLAP)
     /*
      * Win95 has several evil console bugs: one of these is that the last
      * character read using getch() is passed to the next read: this is
-- 
2.6.2

