diff --git a/src/efivar.c b/src/efivar.c
index d4d9fce..ac02aee 100644
--- a/src/efivar.c
+++ b/src/efivar.c
@@ -33,6 +33,10 @@
 #define ACTION_PRINT		0x2
 #define ACTION_APPEND		0x4
 #define ACTION_LIST_GUIDS	0x8
+#define ACTION_OVERWRITE	0x10
+
+#define APPEND	0
+#define WRITE	1
 
 static const char *attribute_names[] = {
 	"Non-Volatile",
@@ -187,7 +191,7 @@ show_variable(char *guid_name)
 }
 
 static void
-append_variable(const char *guid_name, void *data, size_t data_size, int attrib)
+edit_variable(const char *guid_name, void *data, size_t data_size, int attrib, int edit_type)
 {
 	efi_guid_t guid;
 	char *name = NULL;
@@ -204,8 +208,17 @@ append_variable(const char *guid_name, void *data, size_t data_size, int attrib)
 	if (attrib != 0)
 		old_attributes = attrib;
 
-	rc = efi_append_variable(guid, name, data, data_size,
-				old_attributes);
+	switch (edit_type){
+		case APPEND:
+			rc = efi_append_variable(guid, name, data, data_size,
+					old_attributes);
+			break;
+		case WRITE:
+			rc = efi_set_variable(guid, name, data, data_size,
+					old_attributes);
+			break;
+	}
+
 	if (rc < 0) {
 		fprintf(stderr, "efivar: %m\n");
 		exit(1);
@@ -279,6 +292,8 @@ int main(int argc, char *argv[])
 		 "attributes to use on append", "<attributes>" },
 		{"list-guids", 'L', POPT_ARG_VAL, &action,
 		 ACTION_LIST_GUIDS, "show internal guid list", NULL },
+		{"write", 'w', POPT_ARG_VAL, &action,
+		 ACTION_OVERWRITE, "write to variable specified by --name", NULL },
 		POPT_AUTOALIAS
 		POPT_AUTOHELP
 		POPT_TABLEEND
@@ -327,7 +342,13 @@ int main(int argc, char *argv[])
 		case ACTION_APPEND | ACTION_PRINT:
 			validate_name(name);
 			prepare_data(file, &data, &data_size);
-			append_variable(name, data, data_size, attributes);
+			edit_variable(name, data, data_size, attributes, APPEND);
+			break;
+		case ACTION_OVERWRITE | ACTION_PRINT:
+			validate_name(name);
+			prepare_data(file, &data, &data_size);
+			edit_variable(name, data, data_size, attributes, WRITE);
+			break;
 		case ACTION_LIST_GUIDS: {
 			extern struct guidname efi_well_known_guids;
 			extern struct guidname efi_well_known_guids_end;
