diff --git a/src/efivar.c b/src/efivar.c
index a7b473d..9a63b62 100644
--- a/src/efivar.c
+++ b/src/efivar.c
@@ -33,11 +33,15 @@
 #define ACTION_PRINT	0x2
 #define ACTION_APPEND	0x4
 #define ACTION_OVERWRITE	0x10
+#define ACTION_PRINT_DEC	0x20
 #define GUID_FORMAT "%08x-%04x-%04x-%04x-%02x%02x%02x%02x%02x%02x"
 
 #define APPEND	0
 #define WRITE	1
 
+#define VERBOSE	0//default
+#define DECIMAL	1
+
 static const char *attribute_names[] = {
 	"Non-Volatile",
 	"Boot Service Access",
@@ -105,7 +109,7 @@ parse_name(const char *guid_name, char **name, efi_guid_t *guid)
 }
 
 static void
-show_variable(char *guid_name)
+show_variable(char *guid_name, int display_type)
 {
 	efi_guid_t guid;
 	char *name = NULL;
@@ -124,49 +128,67 @@ show_variable(char *guid_name)
 		exit(1);
 	}
 
-	printf("GUID: "GUID_FORMAT "\n",
-		 	guid.a, guid.b, guid.c, bswap_16(guid.d),
-			guid.e[0], guid.e[1], guid.e[2], guid.e[3],
-			guid.e[4], guid.e[5]);
-	printf("Name: \"%s\"\n", name);
-	printf("Attributes:\n");
-	for (int i = 0; attribute_names[i][0] != '\0'; i++) {
-		if(attributes & (1 << i))
-			printf("\t%s\n", attribute_names[i]);
-	}
-	printf("Value:\n");
-
-	uint32_t index = 0;
-	while (index < data_size) {
-		char charbuf[] = "................";
-		printf("%08x  ", index);
-		/* print the hex values, and render the ascii bits into
-		 * charbuf */
-		while (index < data_size) {
-			printf("%02x ", data[index]);
-			if (index % 8 == 7)
-				printf(" ");
-			if (isprint(data[index]))
-				charbuf[index % 16] = data[index];
-			index++;
-			if (index % 16 == 0)
-				break;
+	if(display_type == VERBOSE) {
+		printf("GUID: "GUID_FORMAT "\n",
+				guid.a, guid.b, guid.c, bswap_16(guid.d),
+				guid.e[0], guid.e[1], guid.e[2], guid.e[3],
+				guid.e[4], guid.e[5]);
+		printf("Name: \"%s\"\n", name);
+		printf("Attributes:\n");
+		for (int i = 0; attribute_names[i][0] != '\0'; i++) {
+			if(attributes & (1 << i))
+				printf("\t%s\n", attribute_names[i]);
 		}
+		printf("Value:\n");
 
-		/* If we're above data_size, finish out the line with space,
-		 * and also finish out charbuf with space */
-		while (index >= data_size && index % 16 != 0) {
-			if (index % 8 == 7)
-				printf(" ");
-			printf("   ");
-			charbuf[index % 16] = ' ';
-
-			index++;
-			if (index % 16 == 0)
-				break;
+		uint32_t index = 0;
+		while (index < data_size) {
+			char charbuf[] = "................";
+			printf("%08x  ", index);
+			/* print the hex values, and render the ascii bits into
+			 * charbuf */
+			while (index < data_size) {
+				printf("%02x ", data[index]);
+				if (index % 8 == 7)
+					printf(" ");
+				if (isprint(data[index]))
+					charbuf[index % 16] = data[index];
+				index++;
+				if (index % 16 == 0)
+					break;
+			}
+
+			/* If we're above data_size, finish out the line with space,
+			 * and also finish out charbuf with space */
+			while (index >= data_size && index % 16 != 0) {
+				if (index % 8 == 7)
+					printf(" ");
+				printf("   ");
+				charbuf[index % 16] = ' ';
+
+				index++;
+				if (index % 16 == 0)
+					break;
+			}
+			printf("|%s|\n", charbuf);
+		}
+	}
+	else if(display_type == DECIMAL) {
+		uint32_t index = 0;
+		while (index < data_size) {
+			// print the dec values
+			while (index < data_size) {
+				printf("%d ", data[index]);
+				if (index % 8 == 7)
+					printf(" ");
+				index++;
+				if (index % 16 == 0)
+					break;
+			}
 		}
-		printf("|%s|\n", charbuf);
+		printf("\n");
 	}
+	
 }
 
 static void
@@ -266,6 +288,8 @@ int main(int argc, char *argv[])
 		 "use data from <file>", "<file>" },
 		{"write", 'w', POPT_ARG_VAL, &action,
    		 ACTION_OVERWRITE, "write to variable specified by --name", NULL },
+		{"decimal_print", 'd', POPT_ARG_VAL, &action,
+   		 ACTION_PRINT_DEC, "print variable in decimal values specified by --name", NULL },
 		POPT_AUTOALIAS
 		POPT_AUTOHELP
 		POPT_TABLEEND
@@ -309,7 +333,11 @@ int main(int argc, char *argv[])
 			break;
 		case ACTION_PRINT:
 			validate_name(name);
-			show_variable(name);
+			show_variable(name, VERBOSE);
+			break;
+		case ACTION_PRINT_DEC | ACTION_PRINT:
+			validate_name(name);
+			show_variable(name, DECIMAL);
 			break;
 		case ACTION_APPEND | ACTION_PRINT:
 			validate_name(name);
