From 3f9f3f19bc3c9dbfd0bb9267c30fd7587ddac845 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Fri, 18 Jul 2014 10:04:38 -0500
Subject: [PATCH 15/22] Excluding HAVE_WINSOCK2_H and HAVE_WS2TCPIP_H on
 PharLAP

Upstream-Status: Inappropriate [Pharlap specific]
---
 lib/config-win32.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lib/config-win32.h b/lib/config-win32.h
index 744e022..b775b4a 100644
--- a/lib/config-win32.h
+++ b/lib/config-win32.h
@@ -147,6 +147,7 @@
 /* Define if you have the <winsock.h> header file. */
 #define HAVE_WINSOCK_H 1
 
+#ifndef PHARLAP
 /* Define if you have the <winsock2.h> header file. */
 #ifndef __SALFORDC__
 #define HAVE_WINSOCK2_H 1
@@ -156,7 +157,7 @@
 #ifndef __SALFORDC__
 #define HAVE_WS2TCPIP_H 1
 #endif
-
+#endif PHARLAP
 /* ---------------------------------------------------------------- */
 /*                        OTHER HEADER INFO                         */
 /* ---------------------------------------------------------------- */
-- 
1.9.1

