From d6745c6d362138bb5c1f8fc45d6c28ec6b0116a5 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Fri, 18 Jul 2014 10:06:51 -0500
Subject: [PATCH 16/22] Excluding HAVE_RAND_SCREEN on PharLAP

Upstream-Status: Inappropriate [Pharlap specific]
---
 lib/config-win32.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/config-win32.h b/lib/config-win32.h
index b775b4a..1ec2b94 100644
--- a/lib/config-win32.h
+++ b/lib/config-win32.h
@@ -220,7 +220,9 @@
 #define HAVE_PERROR 1
 
 /* Define if you have the RAND_screen function when using SSL. */
+#ifndef PHARLAP
 #define HAVE_RAND_SCREEN 1
+#endif
 
 /* Define if you have the `RAND_status' function when using SSL. */
 #define HAVE_RAND_STATUS 1
-- 
1.9.1

