From 43aaca70b2b42a6bac901568fd87687815a0f00b Mon Sep 17 00:00:00 2001
From: Alex J Lennon <ajlennon@dynamicdevices.co.uk>
Date: Sat, 31 May 2014 11:13:05 +0100

---
diff --git a/data/config.in b/data/config.in
index 282bc11..c518e1e 100644
--- a/data/config.in
+++ b/data/config.in
@@ -10,13 +10,13 @@
 	<dllmap dll="i:odbc32.dll" target="libiodbc.dylib" os="osx"/>
 	<dllmap dll="oci" target="libclntsh@libsuffix@" os="!windows"/>
 	<dllmap dll="db2cli" target="libdb2_36@libsuffix@" os="!windows"/>
-	<dllmap dll="MonoPosixHelper" target="$mono_libdir/libMonoPosixHelper@libsuffix@" os="!windows" />
+	<dllmap dll="MonoPosixHelper" target="libMonoPosixHelper.so" os="!windows" />
 	<dllmap dll="libmono-btls-shared" target="$mono_libdir/libmono-btls-shared@libsuffix@" os="!windows" />
 	<dllmap dll="i:msvcrt" target="@LIBC@" os="!windows"/>
 	<dllmap dll="i:msvcrt.dll" target="@LIBC@" os="!windows"/>
 	<dllmap dll="sqlite" target="@SQLITE@" os="!windows"/>
 	<dllmap dll="sqlite3" target="@SQLITE3@" os="!windows"/>
-	<dllmap dll="libX11" target="@X11@" os="!windows" />
+	<dllmap dll="libX11" target="libX11.so.6" os="!windows" />
 	<dllmap dll="libgdk-x11-2.0" target="@GDKX11@" os="!windows"/>
 	<dllmap dll="libgdk_pixbuf-2.0" target="libgdk_pixbuf-2.0.so.0" os="!windows"/>
 	<dllmap dll="libgtk-x11-2.0" target="@GTKX11@" os="!windows"/>
@@ -36,8 +36,8 @@
 		<dllentry dll="__Internal" name="MoveMemory" target="mono_win32_compat_MoveMemory"/>
 		<dllentry dll="__Internal" name="ZeroMemory" target="mono_win32_compat_ZeroMemory"/>
 	</dllmap>
-	<dllmap dll="gdiplus" target="@libgdiplus_install_loc@" os="!windows"/>
-	<dllmap dll="gdiplus.dll" target="@libgdiplus_install_loc@"  os="!windows"/>
+	<dllmap dll="gdiplus" target="libgdiplus.so.0" os="!windows"/>
+	<dllmap dll="gdiplus.dll" target="libgdiplus.so.0"  os="!windows"/>
 	<dllmap dll="gdi32" target="@libgdiplus_install_loc@" os="!windows"/>
 	<dllmap dll="gdi32.dll" target="@libgdiplus_install_loc@" os="!windows"/>
 </configuration>
