From 601c2ab70f7097e95a34bbfe328585e17bc0edd9 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Mon, 21 Jul 2014 13:00:51 -0500
Subject: [PATCH 02/22] Including wrn/coreip/hostLib.h in lib/hostip4.c for
 VxWorks builds

Upstream-Status: Inappropriate [NI VxWorks specific change]
---
 lib/hostip4.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/hostip4.c b/lib/hostip4.c
index 3a05d10..fa3d8b7 100644
--- a/lib/hostip4.c
+++ b/lib/hostip4.c
@@ -36,6 +36,10 @@
 #include <inet.h>
 #endif
 
+#ifdef __VXWORKS__
+#include <wrn/coreip/hostLib.h>
+#endif
+
 #ifdef HAVE_PROCESS_H
 #include <process.h>
 #endif
-- 
1.9.1

