From 2506dfdea77c2ab8010e353e465f65d919a9b983 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Mon, 21 Jul 2014 13:40:04 -0500
Subject: [PATCH 03/22] Including selectLib.h in lib/select.c for VxWorks
 builds

Upstream-Status: Inappropriate [NI VxWorks specific change]
---
 lib/select.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/select.c b/lib/select.c
index fc00ce9..14b36c8 100644
--- a/lib/select.c
+++ b/lib/select.c
@@ -23,8 +23,9 @@
 #include "curl_setup.h"
 
 #ifdef __VXWORKS__
-#include <taskLib.h>
-#include <sysLib.h>
+//#include <taskLib.h> DIFF_MARK
+//#include <sysLib.h> DIFF_MARK
+#include <selectLib.h>
 #endif
 
 #ifdef HAVE_SYS_SELECT_H
-- 
1.9.1

