From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jonathan David <jonathan.david@ni.com>
Date: Mon, 21 Jul 2014 15:30:46 -0500
Subject: [PATCH] Made curl_easy_init() thread safe

Upstream-Status: Inappropriate [creates Linux dependency]

diff --git a/lib/easy.c b/lib/easy.c
index b6f4627b6..3b210f077 100644
--- a/lib/easy.c
+++ b/lib/easy.c
@@ -318,14 +318,23 @@ struct Curl_easy *curl_easy_init(void)
   struct Curl_easy *data;
 
   /* Make sure we inited the global SSL stuff */
+#ifdef CURL_THREAD_SAFE_INIT
+  curl_global_mutex_lock();
+#endif
   if(!initialized) {
     result = curl_global_init(CURL_GLOBAL_DEFAULT);
     if(result) {
       /* something in the global init failed, return nothing */
       DEBUGF(fprintf(stderr, "Error: curl_global_init failed\n"));
+#ifdef CURL_THREAD_SAFE_INIT
+      curl_global_mutex_unlock();
+#endif
       return NULL;
     }
   }
+#ifdef CURL_THREAD_SAFE_INIT
+  curl_global_mutex_unlock();
+#endif
 
   /* We use curl_open() with undefined URL so far */
   result = Curl_open(&data);
-- 
2.28.0

