From 03d8bd548872eb9d6838ac2f6090beda40d5513a Mon Sep 17 00:00:00 2001
From: Jonathan David <jonathan.david@ni.com>
Date: Mon, 21 Jul 2014 15:28:15 -0500
Subject: [PATCH] Made curl_global_cleanup() thread safe

Upstream-Status: Inappropriate [creates Linux dependency]
---
 lib/easy.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/lib/easy.c b/lib/easy.c
index 96599ab..ba828e4 100644
--- a/lib/easy.c
+++ b/lib/easy.c
@@ -278,11 +278,24 @@ CURLcode curl_global_init_mem(long flags, curl_malloc_callback m,
  */
 void curl_global_cleanup(void)
 {
+#ifdef CURL_THREAD_SAFE_INIT
+  curl_global_mutex_lock();
+#endif
   if(!initialized)
+  {
+#ifdef CURL_THREAD_SAFE_INIT
+    curl_global_mutex_unlock();
+#endif
     return;
+  }
 
   if(--initialized)
+  {
+#ifdef CURL_THREAD_SAFE_INIT
+    curl_global_mutex_unlock();
+#endif
     return;
+  }
 
   Curl_ssl_cleanup();
   Curl_resolver_global_cleanup();
@@ -300,6 +313,9 @@ void curl_global_cleanup(void)
 #endif
 
   init_flags  = 0;
+#ifdef CURL_THREAD_SAFE_INIT
+  curl_global_mutex_unlock();
+#endif
 }
 
 /*
-- 
2.20.1

