From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Fri, 18 Jul 2014 10:06:51 -0500
Subject: [PATCH] Excluding HAVE_RAND_SCREEN on PharLAP

Upstream-Status: Inappropriate [Pharlap specific]

diff --git a/lib/config-win32.h b/lib/config-win32.h
index 04c25f432..9e6d92296 100644
--- a/lib/config-win32.h
+++ b/lib/config-win32.h
@@ -223,7 +223,9 @@
 #define HAVE_PERROR 1
 
 /* Define if you have the RAND_screen function when using SSL. */
+#ifndef PHARLAP
 #define HAVE_RAND_SCREEN 1
+#endif
 
 /* Define if you have the `RAND_status' function when using SSL. */
 #define HAVE_RAND_STATUS 1
-- 
2.28.0

