From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Richard Tollerton <rich.tollerton@ni.com>
Date: Wed, 30 Sep 2020 17:26:28 -0500
Subject: [PATCH] transfer.c: avoid WSAIoctl() on PHARLAP

Upstream-Status: Inappropriate [Pharlap specific]

diff --git a/lib/transfer.c b/lib/transfer.c
index a07c7af8c..16eb05b38 100644
--- a/lib/transfer.c
+++ b/lib/transfer.c
@@ -974,7 +974,7 @@ CURLcode Curl_done_sending(struct connectdata *conn,
   return CURLE_OK;
 }
 
-#if defined(WIN32) && !defined(USE_LWIPSOCK)
+#if defined(WIN32) && !defined(USE_LWIPSOCK) && !defined(PHARLAP)
 #ifndef SIO_IDEAL_SEND_BACKLOG_QUERY
 #define SIO_IDEAL_SEND_BACKLOG_QUERY 0x4004747B
 #endif
-- 
2.28.0

