diff --git a/grub-core/commands/iorw.c b/grub-core/commands/iorw.c
index a0c164e..49e55ef 100644
--- a/grub-core/commands/iorw.c
+++ b/grub-core/commands/iorw.c
@@ -26,6 +26,7 @@
 
 GRUB_MOD_LICENSE ("GPLv3+");
 
+static grub_extcmd_t cmd_read_bit;
 static grub_extcmd_t cmd_read_byte, cmd_read_word, cmd_read_dword;
 static grub_command_t cmd_write_byte, cmd_write_word, cmd_write_dword;
 
@@ -36,7 +37,6 @@ static const struct grub_arg_option options[] =
     {0, 0, 0, 0, 0, 0}
   };
 
-
 static grub_err_t
 grub_cmd_read (grub_extcmd_context_t ctxt, int argc, char **argv)
 {
@@ -75,6 +75,32 @@ grub_cmd_read (grub_extcmd_context_t ctxt, int argc, char **argv)
 }
 
 static grub_err_t
+grub_cmd_read_bit(grub_extcmd_context_t ctxt, int argc, char **argv)
+{
+  grub_port_t addr;
+  grub_uint32_t index = 0;
+  grub_uint32_t value = 0;
+  grub_uint32_t bit_value = 0;
+  if (argc != 2)
+    return grub_error (GRUB_ERR_BAD_ARGUMENT, N_("two arguments expected: address and bit index"));
+
+  addr = grub_strtoul (argv[0], 0, 0);
+  index  = grub_strtoul (argv[1], 0, 0);
+  value = grub_inb (addr);
+  bit_value = (value & (1 << index)) ? 1 : 0;
+  if (ctxt->state[0].set)
+    {
+      char buf[sizeof ("XXXXXXXX")];
+      grub_snprintf (buf, sizeof (buf), "%d", bit_value);
+      grub_env_set (ctxt->state[0].arg, buf);
+    }
+  else
+    grub_printf ("%d\n", bit_value);
+
+  return 0;
+}
+
+static grub_err_t
 grub_cmd_write (grub_command_t cmd, int argc, char **argv)
 {
   grub_port_t addr;
@@ -130,6 +156,10 @@ GRUB_MOD_INIT(memrw)
     grub_register_extcmd ("inl", grub_cmd_read, 0,
 			  N_("PORT"), N_("Read 32-bit value from PORT."),
 			  options);
+  cmd_read_bit =
+    grub_register_extcmd ("inbit", grub_cmd_read_bit, 0,
+			  N_("PORT"), N_("Read 1-bit value from PORT."),
+			  options);
   cmd_write_byte =
     grub_register_command ("outb", grub_cmd_write,
 			   N_("PORT VALUE [MASK]"),
@@ -149,6 +179,7 @@ GRUB_MOD_FINI(memrw)
   grub_unregister_extcmd (cmd_read_byte);
   grub_unregister_extcmd (cmd_read_word);
   grub_unregister_extcmd (cmd_read_dword);
+  grub_unregister_extcmd (cmd_read_bit);
   grub_unregister_command (cmd_write_byte);
   grub_unregister_command (cmd_write_word);
   grub_unregister_command (cmd_write_dword);
