--- a/iconvdata/gconv-modules	2017-01-26 15:13:45.735807553 -0600
+++ b/iconvdata/gconv-modules	2017-01-26 15:18:04.593091167 -0600
@@ -1136,12 +1136,15 @@
 
 #	from			to			module		cost
 alias	GB13000//		GBK//
-alias	CP936//			GBK//
-alias	MS936//			GBK//
-alias	WINDOWS-936//		GBK//
 module	GBK//			INTERNAL		GBK		1
 module	INTERNAL		GBK//			GBK		1
 
+#       from                    to                      module          cost
+alias	MS936//			CP936//
+alias	WINDOWS-936//		CP936//
+module	CP936//			INTERNAL		CP936		1
+module	INTERNAL		CP936//			CP936		1
+
 #	from			to			module		cost
 module	GBK//			EUC-CN//		GBGBK		1
 module	EUC-CN//		GBK//			GBGBK		1
