From 6e3debd078969bcc78fec330ec7d67c748e2d14f Mon Sep 17 00:00:00 2001
From: Erick Shepherd <erick.shepherd@ni.com>
Date: Fri, 22 May 2026 10:21:10 -0500
Subject: [PATCH] sideband_data: Fix format-security warning in sprintf

Upstream-Status: Pending

Signed-off-by: Erick Shepherd <erick.shepherd@ni.com>
---
 src/sideband_data.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sideband_data.cc b/src/sideband_data.cc
index 5a6bf0c..96ba8ac 100644
--- a/src/sideband_data.cc
+++ b/src/sideband_data.cc
@@ -351,7 +351,7 @@ std::string GetConnectionAddress(::SidebandStrategy strategy)
 //---------------------------------------------------------------------
 int32_t _SIDEBAND_FUNC GetSidebandConnectionAddress(::SidebandStrategy strategy, char address[1024])
 {
-    sprintf(address, GetConnectionAddress(strategy).c_str());
+    sprintf(address, "%s", GetConnectionAddress(strategy).c_str());
     return 0;
 }
 
